### PR TITLE
Feature/embedding api

### PR DESCRIPTION
**Pull request by pandafulmanda**
concept coach being made embeddable for CNX, making it work with BE, moving CC to modal
- [x] fix the modal stuffs.
- [x] include login!

### CODE REVIEW

File: resources/styles/components/concept-coach/base.less [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  .task-breadcrumbs-step {
+    &:hover,
+    &.active {
+      box-shadow: none;      
+    }
+  }
+
+  .concept-coach-login {
+    cursor: pointer;
+    background: url('../../../images/coach.gif');
```
Comment (philschatz) [inline resources/styles/components/concept-coach/base.less]: huh?

File: src/concept-coach/index.cjsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+EventEmitter2 = require 'eventemitter2'
+
+helpers = require '../helpers'
+api = require '../api'
+
+{ModalCoach, channel} = require './modal-coach'
+
+CCWrapped = helpers.wrapComponent(ModalCoach)
+
+publicChannel = new EventEmitter2 wildcard: true
```
Comment (philschatz) [inline src/concept-coach/index.cjsx:11]: why not just extend `class Foo extends EventEmitter2`?

Comment (pandafulmanda) [inline src/concept-coach/index.cjsx:11]: Could I address this in a later clean up pr?

Comment (philschatz) [inline src/concept-coach/index.cjsx:11]: Yep!

**Pull request merged**